packages/bloc/lib/src/bloc_base.dart

```diff
@@ -1,10 +1,46 @@
 part of bloc;
 
+/// An object that provides access to a stream of states over time.
+abstract class Streamable<State> {
+  /// The current [stream] of states.
+  Stream<State> get stream;
+}
+
+/// A [Streamable] that provides synchronous access to the current [state].
+abstract class StateStreamable<State> extends Streamable<State> {
+  /// The current [state].
+  State get state;
+}
+
+/// An object which can emit new states.
+abstract class Emittable<State> {
```

Comment:
This is only style suggestion:
I like type generics like `<State extends Object>` if anything except Null and Void
or `<State extends Object?>` if litteraly anything

Reply:
Agreed but that would be breaking so I would propose differing that change until v9.0.0.